### PR TITLE
Continue working on migrate command

### PR DESCRIPTION
**Pull request by akalburgi**
Adding migrate command to create daml-project.yaml. A small utility to migrate towards newer daml-assistant

### CODE REVIEW

File: da-assistant/DA/Sdk/Cli.hs [hunk 1]
```diff
@@ -289,6 +291,9 @@ handle env configFile action = do
             deprecated $ restart svc
         Studio                 ->
             deprecated $ runJob (JobStudio ".")
+        Migrate -> do
+            res <- liftIO $ copyDetailsFromOldConf proj
+            whenLeft res $ \ex -> throwCliError (CliTextError (DT.pack $ "Could no copy over the details from da.yml" ++ show ex))
```
Comment (fmota-da) [inline da-assistant/DA/Sdk/Cli.hs]:
```suggestion
            whenLeft res $ \ex -> throwCliError (CliTextError (DT.pack $ "Could no copy over the details from da.yaml" ++ show ex))
```

File: da-assistant/DA/Sdk/Cli/Command.hs [hunk 2]
```diff
@@ -114,6 +114,7 @@ actionParser = subparser
   <> subcommand "stop" (Normal . Stop <$> servicesParser) "Stop a given service"
   <> subcommand "feedback" (pure $ Normal SendFeedback) "Send us feedback!"
   <> subcommand "studio" (pure $ Normal Studio) "Start DAML Studio in the current project"
+  <> subcommand "migrate" (pure $ Normal Migrate) "Migrate project config to new daml-assistant"
```
Comment (fmota-da) [inline da-assistant/DA/Sdk/Cli/Command.hs]: Can we move this to the hidden command list for now, as we decide how to do migration?

File: da-assistant/DA/Sdk/Version.hs [hunk 3]
```diff
@@ -77,6 +78,15 @@ showSemVersion (SemVersion major minor patch mbPreRelease) =
...
 
 -- | Parse a semantic version.
+-- | Format a semantic version according to spec compatable for new daml-assitant.
+showSemVersionCompatable :: SemVersion -> T.Text
+showSemVersionCompatable (SemVersion _ minor patch mbPreRelease) =
+    case mbPreRelease of
+        Just preRelease ->
+            format (s%"-"%d%"."%d%"-"%s) "nightly-100" minor patch preRelease
+        Nothing ->
+            format (s%"."%d%"."%d) "nightly-100" minor patch
```
Comment (fmota-da) [inline da-assistant/DA/Sdk/Version.hs]: Great!

File: da-assistant/DA/Sdk/Cli/Job.hs [hunk 4]
```diff
@@ -568,6 +569,15 @@ openResource resourceName = do
...
 
+-- copy details from da.yml to daml-project.daml
+copyDetailsFromOldConf :: Maybe Project -> IO (Either IOException ())
+copyDetailsFromOldConf mbProject = do
+  case mbProject of
+    Just proj -> do
+      let sdkVersion = V.showSemVersionCompatable $ projectSDKVersion proj
+      let versionObject = Y.object [("sdk-version", Y.String sdkVersion)]
+      try $ liftIO (Y.encodeFileWith Y.defaultEncodeOptions "daml.yaml"  versionObject )
+    Nothing -> return $ Left (userError "command has be run from within a project")
```
Comment (fmota-da) [inline da-assistant/DA/Sdk/Cli/Job.hs]:
```suggestion
    Nothing -> return $ Left (userError "Command must be run from within a project.")
```

File: da-assistant/DA/Sdk/Cli/Job.hs [hunk 5]
```diff
@@ -568,6 +569,15 @@ openResource resourceName = do
...
           " \n To access the resource, try pointing the browser to " ++ strRes)
 
+-- copy details from da.yml to daml-project.daml
+copyDetailsFromOldConf :: Maybe Project -> IO (Either IOException ())
+copyDetailsFromOldConf mbProject = do
+  case mbProject of
+    Just proj -> do
+      let sdkVersion = V.showSemVersionCompatable $ projectSDKVersion proj
+      let versionObject = Y.object [("sdk-version", Y.String sdkVersion)]
+      try $ liftIO (Y.encodeFileWith Y.defaultEncodeOptions "daml.yaml"  versionObject )
```
Comment (fmota-da) [inline da-assistant/DA/Sdk/Cli/Job.hs]: Some pesky whitespace:

```suggestion
      try $ liftIO (Y.encodeFileWith Y.defaultEncodeOptions "daml.yaml" versionObject)
```

File: da-assistant/DA/Sdk/Version.hs [hunk 6]
```diff
@@ -77,6 +78,15 @@ showSemVersion (SemVersion major minor patch mbPreRelease) =
             format (d%"."%d%"."%d) major minor patch
 
 -- | Parse a semantic version.
```
Comment (fmota-da) [inline da-assistant/DA/Sdk/Version.hs]: This documentation comment `Parse a semantic version` should be adjacent to the parseSemVersion function, as it was before.

File: da-assistant/DA/Sdk/Version.hs [hunk 7]
```diff
@@ -77,6 +78,15 @@ showSemVersion (SemVersion major minor patch mbPreRelease) =
...
+-- | Format a semantic version according to spec compatable for new daml-assitant.
+showSemVersionCompatable :: SemVersion -> T.Text
```
Comment (fmota-da) [inline da-assistant/DA/Sdk/Version.hs]: Nitpick: it should be spelled `showSemVersionCompatible`

**Comment by akalburgi** [comment]
retest this please

**Pull request merged**